feature-libs/product-configurator/rulebased/components/attribute/types/checkbox-list/configurator-attribute-checkbox-list.component.ts

```diff
@@ -55,11 +55,36 @@ export class ConfiguratorAttributeCheckBoxListComponent
     );
 
     const event: ConfigFormUpdateEvent = {
-      ownerKey: this.ownerKey,
       changedAttribute: {
         ...this.attribute,
         values: selectedValues,
       },
+      ownerKey: this.ownerKey,
+      updateType: Configurator.UpdateType.ATTRIBUTE,
+    };
+
+    this.selectionChange.emit(event);
+  }
+
+  onChangeQuantity(eventObject): void {
+    const value: Configurator.Value = this.configUtilsService
+      .assembleValuesForMultiSelectAttributes(
+        this.attributeCheckBoxForms,
+        this.attribute
+      )
+      .find((item) => item.valueCode === eventObject.valueCode);
+
+    if (!value) return;
```

Comment:
Isn't it an exceptional situation if we don't find any value? I guess we shouldn't just ignore it

Reply:
yes, it is an exception, but i put it for protection against sending an incorrect request

Reply:
Then we should throw an error here (if it can never happen), or we should at least write a log message (in development mode only) if it can happen in exceptional situations

Reply:
Otherwise, in case the update is not done, error analysis would be quite difficult from my point of view

Reply:
add console on devMode